webapp/tests/karma/ts/effects/contacts.effects.spec.ts

```diff
@@ -0,0 +1,234 @@
+      effects.setSelectedContact.subscribe();
+
+      expect(settingSelected.callCount).to.equal(1);
+      expect(setLoadingSelectedContact.callCount).to.equal(1);
+      expect(setContactsLoadingSummary.callCount).to.equal(1);
+      expect(clearCancelCallback.callCount).to.equal(1);
+      expect(contactViewModelGeneratorService.loadChildren.callCount).to.equal(1);
+    });
+
+    it('should catch loadChildren errors', fakeAsync(() => {
+      contactViewModelGeneratorService.loadChildren.rejects({ error: 'we have a problem'});
+      actions$ = of(ContactActionList.setSelectedContact({ _id: 'contactid', doc: {} }));
+      effects.setSelectedContact.subscribe();
+      flush();
+
+      expect(contactViewModelGeneratorService.loadChildren.callCount).to.equal(1);
+      expect(unsetSelected.callCount).to.equal(1);
+    }));
+  });
+
+  describe('receiveSelectedContactReports', () => {
+    let unsetSelected;
+    let receiveSelectedContactReports;
+
+    beforeEach(() => {
+      unsetSelected = sinon.stub(GlobalActions.prototype, 'unsetSelected');
+      receiveSelectedContactReports = sinon.stub(ContactsActions.prototype, 'receiveSelectedContactReports');
+    });
+
+    it('should catch loadReports errors', fakeAsync(() => {
+      contactViewModelGeneratorService.loadReports.rejects({ error: 'we have a problem'});
+      actions$ = of(ContactActionList.receiveSelectedContactChildren([]));
+      effects.receiveSelectedContactReports.subscribe();
+      flush();
+
+      expect(contactViewModelGeneratorService.loadReports.callCount).to.equal(1);
+      expect(unsetSelected.callCount).to.equal(1);
+    }));
+
+    it('should caall the receiveSelectedContactReports action', fakeAsync(() => {
+      actions$ = of(ContactActionList.receiveSelectedContactChildren([]));
+      effects.receiveSelectedContactReports.subscribe();
+      flush();
+
+      expect(contactViewModelGeneratorService.loadReports.callCount).to.equal(1);
+      expect(receiveSelectedContactReports.callCount).to.equal(1);
+    }));
+  });
+
+  describe('updateSelectedContactSummary', () => {
+    let updateSelectedContactsTasks;
+    let setContactsLoadingSummary;
+    let updateSelectedContactSummary;
+    let unsetSelected;
+
+    beforeEach(() => {
+      updateSelectedContactsTasks = sinon.stub(ContactsActions.prototype, 'updateSelectedContactsTasks');
+      setContactsLoadingSummary = sinon.stub(ContactsActions.prototype, 'setContactsLoadingSummary');
+      updateSelectedContactSummary = sinon.stub(ContactsActions.prototype, 'updateSelectedContactSummary');
+      unsetSelected = sinon.stub(GlobalActions.prototype, 'unsetSelected');
+    });
+
+    it('should call the right actions', fakeAsync(() => {
+      actions$ = of(ContactActionList.receiveSelectedContactReports([]));
+      effects.updateSelectedContactSummary.subscribe();
+      flush();
+
+      expect(updateSelectedContactSummary.callCount).to.equal(1);
```

Comment:
can we check that updateSelectedContactSummary and updateSelectedContactsTasks are receiving the right data when being called?